assessment/collectors/infoschema_collector.go

```diff
@@ -131,28 +140,79 @@ func (c InfoSchemaCollector) ListColumns() map[string][]string {
 	return columnNames
 }
 
-func (c InfoSchemaCollector) ListIndexesAndTypes() map[string]string {
-	indexes := make(map[string]string)
+func (c InfoSchemaCollector) ListIndexes() (map[string]utils.SrcIndexDetails, map[string]utils.SpIndexDetails) {
+	srcIndexes := make(map[string]utils.SrcIndexDetails)
+	spIndexes := make(map[string]utils.SpIndexDetails)
 	for i := range c.indexes {
-		indexes[c.indexes[i].Name] = c.indexes[i].Ty
+		srcIndexes[c.indexes[i].IndexDef.Id] = utils.SrcIndexDetails{
+			Id:        c.indexes[i].IndexDef.Id,
+			Name:      c.indexes[i].IndexDef.Name,
+			TableId:   c.indexes[i].TableId,
+			Type:      c.indexes[i].Ty,
+			TableName: c.conv.SrcSchema[c.indexes[i].TableId].Name,
+			IsUnique:  c.indexes[i].IndexDef.Unique,
+		}
+		spIndexes[c.indexes[i].IndexDef.Id] = utils.SpIndexDetails{
+			Id:        c.indexes[i].IndexDef.Id,
+			Name:      internal.ToSpannerIndexName(c.conv, c.indexes[i].IndexDef.Name),
+			TableId:   c.indexes[i].TableId,
+			IsUnique:  c.indexes[i].IndexDef.Unique,
+			TableName: c.conv.SpSchema[c.indexes[i].TableId].Name,
+		}
 	}
-	return indexes
+	return srcIndexes, spIndexes
 }
 
-func (c InfoSchemaCollector) ListTriggers() []utils.TriggerAssessmentOutput {
-	var triggersAssessmentOutput []utils.TriggerAssessmentOutput
+func (c InfoSchemaCollector) ListTriggers() map[string]utils.TriggerAssessmentOutput {
```

Comment:
can you clarify in comments what the key and values are expected to be where maps are being used ?